### PR TITLE
add search filter to datasets route

### PR DESCRIPTION
**Pull request by merobi-hub**
Screenshot 2023-06-08 at 16 20 43

### Problem

Currently there is no mechanism for filtering datasets on the datasets route. In the case of deployments with many datasets, it can be difficult to find a specific dataset.

### Solution

This adds an input, corresponding changes to the `Datasets` class, and a new `SearchState` interface to add a dynamic search filter based on dataset names.  

> **Note:** All database schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

One-line summary: adds a search filter to the datasets route

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [x] You've included a one-line summary of your change for the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) (_Depending on the change, this may not be necessary_).
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [ ] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

File: web/src/routes/datasets/Datasets.tsx [hunk 1]
```diff
@@ -71,9 +96,21 @@ class Datasets extends React.Component<DatasetsProps> {
...
+                </Box>
+                <Box>
+                  <MqInputBase 
+                    type='text' 
+                    value={this.state.searchTerm}
+                    placeholder='Filter by name' 
+                    id={'filterBar'} 
+                    onChange={string => this.handleSearch(string)}
+                  />
+                </Box>
```
Comment (phixMe) [inline web/src/routes/datasets/Datasets.tsx:113]: Can you put this in a flex box with 'justify-content': 'space-between', 'align-items': 'center' to make it a single line with datasets.

File: web/src/routes/datasets/Datasets.tsx [hunk 2]
```diff
@@ -34,9 +44,24 @@ interface DispatchProps {
...
-class Datasets extends React.Component<DatasetsProps> {
+class Datasets extends React.Component<DatasetsProps, SearchState> {
+  constructor(props: DatasetsProps) {
+    super(props)
+    this.state = {
+      searchTerm: ''
+    }
+  }
+  
+  handleSearch(event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement> ) {
```
Comment (phixMe) [inline web/src/routes/datasets/Datasets.tsx:61]: I think it's only an inputElement here, no need for the search area.

Comment (merobi-hub) [reply web/src/routes/datasets/Datasets.tsx:61]: @phixMe When I remove HTMLTextAreaElement from the type, it doesn't compile and I get this error msg: `Argument of type 'ChangeEvent<HTMLInputElement | HTMLTextAreaElement>' is not assignable to parameter of type 'ChangeEvent<HTMLInputElement>'.` Can you advise?

**Comment by wslulciuc** [comment]
@merobi-hub do we still want to merge this PR? /cc @phixMe

**Comment by phixMe** [comment]
Let's close this for now because we have some new backend work in progress that would help improve this feature.

**Pull request closed without merging**